webpack/assets/javascripts/foreman_tools.test.js

```diff
@@ -48,19 +48,6 @@ describe('activateDatatables', () => {
   });
 });
 
-describe('activateTooltips', () => {
-  it('calls $.fn.tooltip on all matching elements', () => {
-    const elements = `<div rel='twipsy'></div>
-      <div class='ellipsis'></div>
-      <div title='test'></div>
-      <div title='test' rel='popover'></div>`;
-
-    $.fn.tooltip = jest.fn();
-    tools.activateTooltips(elements);
-    expect($.fn.tooltip).toHaveBeenCalledTimes(3);
-  });
-});
```

Comment:
since the tooltips are only initialized once now and not for each element this test no longer makes sense.

Reply:
> Instead of iterating over all elements with tooltips or popovers and
> initializing each one seperately, initialize them once with a selector
> to identify which elements should have them. On pages with hundreds or
> thousands of tooltips and popovers this could have a significant
> performance impact on the page rendering.

Some more backgrounds:
Our Puppet / Foreman environment consists of 5  puppet environments, ~600 (sub-)class, ~4700 smart class parameters and ~ 1800 hosts. We use Foreman as ENC, so there are a lot of overrides.
Foreman is running on 4 nodes, each with 4CPUs and 10G Memory.

Without this fix it can take up to 5 minutes to load the edit page of one puppetclass (if you have ~3000 overrides). The majority of the waiting time the browser (chrome 88) was rendering the java script stack (cpu usage == 100%). 

In our futher test case we used a class with ~ 1100 overrides. Without this fix it took ~55s to load the edit page, with this fix only 6-8s :) :) Thanks to @tbrisker :)
More Details, logs, discussion can be found at https://community.theforeman.org/t/foreman-2-2-2-edit-puppet-classes-and-smart-class-parameters-extrem-slow/22399